trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid.F

```diff
@@ -0,0 +1,371 @@
+                write(log_warning,*) "Warning, initial storage is less than zero for reservoir ", lake_number_string, &
+                ". Setting initial storage to zero."
+
+            end if
+
+            ! Allocate a single level pool reservoir
+            allocate(levelpool_type :: this%state%levelpool_ptr)
+
+            ! Initialize level pool reservoir
+            call this%state%levelpool_ptr%init(water_elevation, lake_area, &
+            weir_elevation, weir_coeffecient, weir_length, orifice_elevation, &
+            orifice_coefficient, orifice_area, lake_max_water_elevation, lake_number)
+
+        end if
+
+    end subroutine hybrid_type_init
+
+
+    ! Hybrid Destructor
+    subroutine hybrid_type_destroy(this)
+        implicit none
+        class(persistence_levelpool_hybrid_type), intent(inout) :: this ! object being destroyed
+    end subroutine hybrid_type_destroy
+
+
+    ! Subroutine for running reservoir for a hybrid reservoir
+    subroutine run_hybrid_reservoir(this, previous_timestep_inflow, inflow, &
+        lateral_inflow, water_elevation, outflow, routing_period)
+        implicit none
+        class(persistence_levelpool_hybrid_type), intent(inout) :: this
+        real, intent(in)    :: previous_timestep_inflow ! cubic meters per second (cms)
+        real, intent(in)    :: inflow                   ! cubic meters per second (cms)
+        real, intent(in)    :: lateral_inflow           ! cubic meters per second (cms)
+        real, intent(inout) :: water_elevation          ! meters
+        real, intent(out)   :: outflow                  ! cubic meters per second (cms)
+        real, intent(in)    :: routing_period           ! seconds
+        real                :: delta_storage            ! timestep change in storage (cubic meters)
+        real                :: local_water_elevation    ! water elevation passed to levelpool (meters AMSL)
+        real                :: levelpool_outflow        ! cubic meters per second (cms)
+        logical             :: max_storage_reached      ! flag for when max storage is reached
+        integer             :: gage_lookback_seconds    ! seconds between current model
+                                                        ! time and the time a gage
+                                                        ! discharge is read
+
+        max_storage_reached = .false.
+        gage_lookback_seconds = 0
+
+        ! Update input variables
+        this%input%inflow = inflow
+        this%input%lateral_inflow = lateral_inflow
+
+        ! The initialization and management of water elevation might diverge between the global state of the model
+        ! and the internal state. If water elevation is refactored to no longer be a global state, then this check
+        ! is not necessary. This currently allows restarts to work without the reservoir managing its own restart.
+        if (this%state%water_elevation .ne. water_elevation) then
```

Comment:
Putting this here as a place holder: 
>Essentially, there is a water elevation managed external to this module that is passed in every time step. We have to make sure it stays consistent within the module every time step. This is to help guard against some external piece of code adjusting the water elevation and then our internal storage/water elevation is no long synchronized.
This also allows us to borrow from the existing restart capability by initializing reservoirs to some default storage/water elevation, then waiting for the first time step to receive the initial condition

Reply:
I just copied this comment from an outdated discussion. It could be incoprorated into the code comment to help clarify it further.

Reply:
@jdmattern-noaa Updated and clarified the description.